cypress/e2e/trafficlight/register.spec.ts

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+function recurse() {
+    cy.log("Requesting next action...");
+    const pollUrl = Cypress.env("TRAFFICLIGHT_URL") + "/client/" + Cypress.env("TRAFFICLIGHT_UUID") + "/poll";
+    const respondUrl = Cypress.env("TRAFFICLIGHT_URL") + "/client/" + Cypress.env("TRAFFICLIGHT_UUID") + "/respond";
+    cy.request(pollUrl).then((resp) => {
+        expect(resp.status).to.eq(200);
+        // promote response out of the callback for future use.
+        const data = resp.body.data;
+        const action = resp.body.action;
+        cy.log("... got ", action, data);
+        switch (action) {
+            case "register":
+                cy.get(".mx_ServerPicker_change", { timeout: 15000 }).click();
+                cy.get(".mx_ServerPickerDialog_continue").should("be.visible");
+                cy.get(".mx_ServerPickerDialog_otherHomeserver").type(data['homeserver_url']['local']);
+                cy.get(".mx_ServerPickerDialog_continue").click();
+                // wait for the dialog to go away
+                cy.get('.mx_ServerPickerDialog').should('not.exist');
+                cy.get("#mx_RegistrationForm_username").should("be.visible");
+                // Hide the server text as it contains the randomly allocated Synapse port
+                cy.get("#mx_RegistrationForm_username").type(data['username']);
+                cy.get("#mx_RegistrationForm_password").type(data['password']);
+                cy.get("#mx_RegistrationForm_passwordConfirm").type(data['password']);
+                cy.get(".mx_Login_submit").click();
+                cy.get('.mx_UseCaseSelection_skip > .mx_AccessibleButton').click();
+                cy.request('POST', respondUrl, { response: "registered" }).then((response) => {
+                    expect(response.status).to.eq(200);
+                });
+                break;
+            case "login":
+                cy.visit("http://localhost:8083/#/login");
```

Comment:
This port probably shouldn't be hardcoded, and should likely match the one in Cypress' config so that cypress does live checking on it for timing purposes

Reply:
Cypress does the right thing if you leave off the host/port - so doing that in bdf4d9d723047bc1c17d11fbcd263ff62a50d025